### PR TITLE
deltaY: fixed model, rendering, tests

### PR DESCRIPTION
**Pull request by denim2x**
- improvements over original model
- fixed decoration rendering and '_extendStyles'
-

### CODE REVIEW

File: src/shapes/text.class.js [hunk 1]
```diff
@@ -1322,30 +1346,37 @@
...
+            );
             boxStart = charBox.left;
             boxWidth = charBox.width;
             lastDecoration = currentDecoration;
             lastFill = currentFill;
-          }
-          else {
+            size = _size;
+            dy = _dy;
+          } else {
```
Comment (asturur) [inline src/shapes/text.class.js]: the linter will error with inline else. it want them on the next line.

File: test/unit/text.js [hunk 2]
```diff
@@ -382,48 +382,48 @@
...
+    equal(text.styles[0][2].fontSize, styleFontSize * schema.size, 'character 2: fontSize has been decreased');
+    equal(text.styles[0][2].deltaY, styleDeltaY + styleFontSize * schema.baseline, 'character 2: deltaY has been decreased');
   });
 
-  test('text subscript - single char', function() {
+  test('text subscript', function() {
     var text = new fabric.Text('xxx', { styles: {
       0: { 0: { stroke: 'black', fill: 'blue' }, 1:  { fill: 'blue' }, 2:  { fontSize: 4, deltaY: 20 }}
     } });
+    ok(typeof text.setSuperscript === 'function');
```
Comment (asturur) [inline test/unit/text.js]: this is setSubscript

File: src/shapes/text.class.js [hunk 3]
```diff
@@ -1320,32 +1345,40 @@
...
-          currentDecoration = this.getValueOfPropertyAt(i, j, type);
-          currentFill = this.getValueOfPropertyAt(i, j, 'fill');
-          if ((currentDecoration !== lastDecoration || currentFill !== lastFill) && boxWidth > 0) {
-            ctx.fillStyle = lastFill;
-            lastDecoration && lastFill && ctx.fillRect(
+          _decoration = this.getValueOfPropertyAt(i, j, type);
+          _fill = this.getValueOfPropertyAt(i, j, 'fill');
+          _size = this.getHeightOfChar(i, j);
+          _dy = this.getValueOfPropertyAt(i, j, 'deltaY');
+          if ((_decoration !== decoration || _fill !== fill || _size !== size || _dy !== dy) && boxWidth > 0) {
```
Comment (denim2x) [inline src/shapes/text.class.js]: had to shorten variables' names to bring this line below 120 chars

Comment (asturur) [reply src/shapes/text.class.js]: just break the line, is valid for eslint. you have to leave the boolean operator a the end of the line.

npm run lint

will give you results.

Comment (denim2x) [reply src/shapes/text.class.js]: ok

**Comment by denim2x** [comment]
done

**Comment by denim2x** [comment]
done

**Pull request merged**